src/modules/Bootloader.rb

```diff
@@ -422,7 +422,22 @@ def ReadOrProposeIfNeeded
         Propose()
       else
         progress_orig = Progress.set(false)
+        if Mode.update
```

Comment:
please also check Stage.initial, as it can be broken in autoupgrade in second stage.